pkg/encoding/basic.go

```diff
@@ -0,0 +1,141 @@
+package encoding
+
+import (
+	"fmt"
+	"math/big"
+	"sort"
+	"strings"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+const separator string = "|"
+
+var (
+	ErrUpkeepKeyNotParsable = fmt.Errorf("upkeep key not parsable")
+)
+
+type BasicEncoder struct{}
+
+// MakeUpkeepKey creates a new types.UpkeepKey from a types.BlockKey and a types.UpkeepIdentifier
+func (kb BasicEncoder) MakeUpkeepKey(blockKey ocr2keepers.BlockKey, upkeepIdentifier ocr2keepers.UpkeepIdentifier) ocr2keepers.UpkeepKey {
+	return ocr2keepers.UpkeepKey(fmt.Sprintf("%s%s%s", blockKey, separator, string(upkeepIdentifier)))
+}
+
+// SplitUpkeepKey splits a types.UpkeepKey into its constituent types.BlockKey and types.UpkeepIdentifier parts
+func (kb BasicEncoder) SplitUpkeepKey(upkeepKey ocr2keepers.UpkeepKey) (ocr2keepers.BlockKey, ocr2keepers.UpkeepIdentifier, error) {
+	if upkeepKey == nil {
+		return "", nil, fmt.Errorf("%w: missing data in upkeep key", ErrUpkeepKeyNotParsable)
+	}
+
+	components := strings.Split(string(upkeepKey), separator)
+	if len(components) != 2 {
+		return "", nil, fmt.Errorf("%w: missing data in upkeep key", ErrUpkeepKeyNotParsable)
+	}
+
+	return ocr2keepers.BlockKey(components[0]), ocr2keepers.UpkeepIdentifier(components[1]), nil
+}
+
+// ValidateUpkeepKey returns true if the types.UpkeepKey is valid, false otherwise
+func (kb BasicEncoder) ValidateUpkeepKey(key ocr2keepers.UpkeepKey) (bool, error) {
```

Comment:
moving the comment here about validation of upper bounds on block key and upkeep id